### ISSUE TITLE
Segfault when using Imagick with phpbrew

### ISSUE THREAD

**Issue by bahag-schlachterk** [Imagick/imagick#609]
After an upgrade to Ubuntu 23.04 and starting to rebuild php 8.1 or 8.2 by using phpbrew the build works but the resulting version of the extension is broken and will cause php to crash with a seg fault.

Information gathered so far:
- phpbrew used: 2.1
- Imagick used: 6.9.11-60 Q16 x86_64 2021-01-25 (from ubuntu repo for 23.04, see below)
- Installed packages: 
-- imagemagick-6-common/lunar,lunar,now 8:6.9.11.60+dfsg-1.6
-- imagemagick-6.q16/lunar,now 8:6.9.11.60+dfsg-1.6
-- imagemagick-common/lunar,lunar,now 8:6.9.11.60+dfsg-1.6
-- imagemagick/lunar,now 8:6.9.11.60+dfsg-1.6
-- libimage-magick-perl/lunar,lunar,now 8:6.9.11.60+dfsg-1.6
-- libimage-magick-q16-perl/lunar,now 8:6.9.11.60+dfsg-1.6
-- libmagick++-6.q16-8/lunar,now 8:6.9.11.60+dfsg-1.6
-- libmagickcore-6-arch-config/lunar,now 8:6.9.11.60+dfsg-1.6
-- libmagickcore-6-headers/lunar,lunar,now 8:6.9.11.60+dfsg-1.6
-- libmagickcore-6.q16-6-extra/lunar,now 8:6.9.11.60+dfsg-1.6
-- libmagickcore-6.q16-6/lunar,now 8:6.9.11.60+dfsg-1.6
-- libmagickcore-6.q16-dev/lunar,now 8:6.9.11.60+dfsg-1.6
-- libmagickwand-6-headers/lunar,lunar,now 8:6.9.11.60+dfsg-1.6
-- libmagickwand-6.q16-6/lunar,now 8:6.9.11.60+dfsg-1.6
-- libmagickwand-6.q16-dev/lunar,now 8:6.9.11.60+dfsg-1.6
-- libmagickwand-dev/lunar,lunar,now 8:6.9.11.60+dfsg-1.6

Steps to reproduce (as far as known ...):
- Install / Upgrade to Ubuntu 23.04
- install phpbrew 2.10 as recommended
- install / compile a version of php 8.2.x or php 8.1.x with phpbrew including desired variants
- switch or use or call binary for specific version
- run php -v or php -i to see if version is selected and working => works fine
- add imagick to running variant by 'phpbrew ext install imagick'
- compile runs once imagick-dev and other dependencies are installed on host
- run php -v or php -i to check everything still works => get segfault message => disabling the extension needs to be done manually as the activated php is not usable anymore ...

If more information is required, feel free to ask.

**Comment by bahag-schlachterk** [comment]
OK reran with valgrind:
`$ valgrind php -v
==272669== Memcheck, a memory error detector
==272669== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==272669== Using Valgrind-3.19.0 and LibVEX; rerun with -h for copyright info
==272669== Command: php -v
==272669== 
==272669== Invalid read of size 8
==272669==    at 0x908AADE: ??? (in /usr/lib/x86_64-linux-gnu/libgomp.so.1.0.0)
==272669==    by 0x400533D: call_init.part.0 (dl-init.c:70)
==272669==    by 0x4005427: call_init (dl-init.c:33)
==272669==    by 0x4005427: _dl_init (dl-init.c:117)
==272669==    by 0x4001561: _dl_catch_exception (dl-catch.c:211)
==272669==    by 0x400C345: dl_open_worker (dl-open.c:808)
==272669==    by 0x400C345: dl_open_worker (dl-open.c:771)
==272669==    by 0x40014DC: _dl_catch_exception (dl-catch.c:237)
==272669==    by 0x400C6BB: _dl_open (dl-open.c:884)
==272669==    by 0x5303BEB: dlopen_doit (dlopen.c:56)
==272669==    by 0x40014DC: _dl_catch_exception (dl-catch.c:237)
==272669==    by 0x4001602: _dl_catch_error (dl-catch.c:256)
==272669==    by 0x53036BE: _dlerror_run (dlerror.c:138)
==272669==    by 0x5303CA0: dlopen_implementation (dlopen.c:71)
==272669==    by 0x5303CA0: dlopen@@GLIBC_2.34 (dlopen.c:81)
==272669==  Address 0x0 is not stack'd, malloc'd or (recently) free'd
==272669== 
==272669== 
==272669== Process terminating with default action of signal 11 (SIGSEGV)
==272669==  Access not within mapped region at address 0x0
==272669==    at 0x908AADE: ??? (in /usr/lib/x86_64-linux-gnu/libgomp.so.1.0.0)
==272669==    by 0x400533D: call_init.part.0 (dl-init.c:70)
==272669==    by 0x4005427: call_init (dl-init.c:33)
==272669==    by 0x4005427: _dl_init (dl-init.c:117)
==272669==    by 0x4001561: _dl_catch_exception (dl-catch.c:211)
==272669==    by 0x400C345: dl_open_worker (dl-open.c:808)
==272669==    by 0x400C345: dl_open_worker (dl-open.c:771)
==272669==    by 0x40014DC: _dl_catch_exception (dl-catch.c:237)
==272669==    by 0x400C6BB: _dl_open (dl-open.c:884)
==272669==    by 0x5303BEB: dlopen_doit (dlopen.c:56)
==272669==    by 0x40014DC: _dl_catch_exception (dl-catch.c:237)
==272669==    by 0x4001602: _dl_catch_error (dl-catch.c:256)
==272669==    by 0x53036BE: _dlerror_run (dlerror.c:138)
==272669==    by 0x5303CA0: dlopen_implementation (dlopen.c:71)
==272669==    by 0x5303CA0: dlopen@@GLIBC_2.34 (dlopen.c:81)
==272669==  If you believe this happened as a result of a stack
==272669==  overflow in your program's main thread (unlikely but
==272669==  possible), you can try to increase the size of the
==272669==  main thread stack using the --main-stacksize= flag.
==272669==  The main thread stack size used in this run was 8388608.
==272669== 
==272669== HEAP SUMMARY:
==272669==     in use at exit: 1,105,972 bytes in 6,392 blocks
==272669==   total heap usage: 7,187 allocs, 795 frees, 1,359,342 bytes allocated
==272669== 
==272669== LEAK SUMMARY:
==272669==    definitely lost: 4,448 bytes in 139 blocks
==272669==    indirectly lost: 0 bytes in 0 blocks
==272669==      possibly lost: 719,088 bytes in 4,998 blocks
==272669==    still reachable: 382,436 bytes in 1,255 blocks
==272669==         suppressed: 0 bytes in 0 blocks
==272669== Rerun with --leak-check=full to see details of leaked memory
==272669== 
==272669== For lists of detected and suppressed errors, rerun with: -s
==272669== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
`

Looking at this stack-trace, I think there might be some issues that come together:
https://github.com/Imagick/imagick/issues/295 looks like the same sort of issue.
Looking at the version of gcc on the new system compared to a system still on Ubuntu 22.10 the difference is:

old 12.2.0-3ubunut1 <=> 12.2.0-17ubunut1 new

Regarding your question about "phpbrew": Take a look at: https://github.com/phpbrew/phpbrew
Short version of it: its a collection of commonly used php-combinations which allows you to build "parallel" installations with specific configs, which is quite handy if you need to try out a version that is not supported by your system natively yet or anymore. So this might also be a point were adjustments of the build config might be applied to work around the issue.

**Comment by Danack** [comment]
> Regarding your question about "phpbrew": Take a look at

Thanks, no. I have to put a limit on what downstream (of this repo) stuff I learn about as it would be too much of a burden otherwise.

> https://github.com/Imagick/imagick/issues/295 looks like the same sort of issue.

Actually, that's a different one. It was (as far as I can understand) a race condition when applications shutdown.

The issue you have appears to be on application start up...

As the issue is in libgomp, which is the open parallel processing library, you could try disabling that by compiling ImageMagick with `--disable-openmp`.

I'll comment on the other issue, but there's not going to be much I can do here.

**Comment by bahag-spiesj** [comment]
Hi there!
At least for me this didn't solve the problem.
According to https://github.com/phpbrew/phpbrew/issues/821 (other extension) I installed imagick with the recommended parameter but still get the segmentation fault error
```shell
~$ phpbrew ext install imagick -- --disable-openmp
``` 
<details><summary>Details</summary>
<p>

~$ phpbrew ext install imagick -- --disable-openmp
Downloading https://pecl.php.net/rest/r/imagick/stable.txt via curl extension
[==================================================================] 5.00/5.00B 100%
Downloading https://pecl.php.net/rest/r/imagick/3.7.0.xml via curl extension
[==================================================================] 842.00/842.00B 100%
Downloading https://pecl.php.net/rest/r/imagick/stable.txt via curl extension
[==================================================================] 5.00/5.00B 100%
Downloading https://pecl.php.net/rest/r/imagick/3.7.0.xml via curl extension
[==================================================================] 842.00/842.00B 100%
Downloading https://pecl.php.net/get/imagick-3.7.0.tgz via curl extension
[==================================================================] 351.70/351.70KB 100%
===> Extracting to /.phpbrew/build/php-8.1.18/ext...
[ ] imagick extension is already disabled for SAPI .
===> Installing imagick extension...
Log stored at: ~/.phpbrew/build/php-8.1.18/ext/imagick/build.log
Changing directory to ~/.phpbrew/build/php-8.1.18/ext/imagick
===> Phpize...
===> Configuring...
===> Building...
===> Running make all: /usr/bin/make -C '~/.phpbrew/build/php-8.1.18/ext/imagick' 'all'  >> '~/.phpbrew/build/php-8.1.18/ext/imagick/build.log' 2>&1
===> Installing...
===> Running make install: /usr/bin/make -C '~/.phpbrew/build/php-8.1.18/ext/imagick' 'install'  >> '~/.phpbrew/build/php-8.1.18/ext/imagick/build.log' 2>&1
===> Extension is installed.
===> Creating config file ~/.phpbrew/php/php-8.1.18/var/db/imagick.ini
===> Enabling extension imagick
[*] imagick extension is enabled for SAPI cli.
Done.
~$ php -v
Segmentation fault (core dumped)

</p>
</details>

**Comment by bahag-schlachterk** [comment]
I opened a bug for the openmp lib:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=109634
According to the investigation there, it is related how the gomp lib is used / called.
Question is: were do we get "off-track":
- the way glibc calls openmp?
- the way imagick calls openmp (possibly via glibc?)
- building / linking while building via phpbrew?

**Comment by Danack** [comment]
Before I do any more on this issue:

Does your company sponsor either Imagick or ImageMagick?

**Comment by bahag-schlachterk** [comment]
Unfortunately we do not sponsor neither of the projects for the time being.

**Comment by nariox** [comment]
Adding `Environment=LD_PRELOAD=libgomp.so.1` to my php-fpm.service "solved" the problem, but it seems like the problem is in ImageMagick rather than Imagick.

**Comment by valerioa** [comment]
I have the same problem on a Ubuntu 22.04.3. I too see the crash in libgomp. libgomp is linked in imagick.so

```
ldd .libs/imagick.so
	linux-vdso.so.1 (0x00007ffcd9df1000)
	libgomp.so.1 => /usr/local/pkg/gcc-13.2.0/lib/../lib64/libgomp.so.1 (0x00007ff094555000)
	libMagickWand-6.Q16.so.6 => /lib/x86_64-linux-gnu/libMagickWand-6.Q16.so.6 (0x00007ff094412000)
	libMagickCore-6.Q16.so.6 => /lib/x86_64-linux-gnu/libMagickCore-6.Q16.so.6 (0x00007ff094146000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007ff093f1e000)
	libX11.so.6 => /lib/x86_64-linux-gnu/libX11.so.6 (0x00007ff093dde000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007ff093cf5000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007ff093cd5000)
	liblcms2.so.2 => /lib/x86_64-linux-gnu/liblcms2.so.2 (0x00007ff093c73000)
	liblqr-1.so.0 => /lib/x86_64-linux-gnu/liblqr-1.so.0 (0x00007ff093a00000)
	libfftw3.so.3 => /lib/x86_64-linux-gnu/libfftw3.so.3 (0x00007ff0937e5000)
	libxml2.so.2 => /lib/x86_64-linux-gnu/libxml2.so.2 (0x00007ff093603000)
	libfontconfig.so.1 => /lib/x86_64-linux-gnu/libfontconfig.so.1 (0x00007ff093c27000)
	libfreetype.so.6 => /lib/x86_64-linux-gnu/libfreetype.so.6 (0x00007ff09353b000)
	libXext.so.6 => /lib/x86_64-linux-gnu/libXext.so.6 (0x00007ff093c12000)
	libbz2.so.1.0 => /lib/x86_64-linux-gnu/libbz2.so.1.0 (0x00007ff093528000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007ff09350c000)
	libltdl.so.7 => /lib/x86_64-linux-gnu/libltdl.so.7 (0x00007ff093501000)
	/lib64/ld-linux-x86-64.so.2 (0x00007ff094616000)
	libxcb.so.1 => /lib/x86_64-linux-gnu/libxcb.so.1 (0x00007ff0934d7000)
	libglib-2.0.so.0 => /lib/x86_64-linux-gnu/libglib-2.0.so.0 (0x00007ff09339d000)
	libicuuc.so.70 => /lib/x86_64-linux-gnu/libicuuc.so.70 (0x00007ff0931a2000)
	liblzma.so.5 => /lib/x86_64-linux-gnu/liblzma.so.5 (0x00007ff093177000)
	libexpat.so.1 => /lib/x86_64-linux-gnu/libexpat.so.1 (0x00007ff093146000)
	libuuid.so.1 => /lib/x86_64-linux-gnu/libuuid.so.1 (0x00007ff09313d000)
	libpng16.so.16 => /lib/x86_64-linux-gnu/libpng16.so.16 (0x00007ff093102000)
	libbrotlidec.so.1 => /lib/x86_64-linux-gnu/libbrotlidec.so.1 (0x00007ff0930f4000)
	libXau.so.6 => /lib/x86_64-linux-gnu/libXau.so.6 (0x00007ff0930ee000)
	libXdmcp.so.6 => /lib/x86_64-linux-gnu/libXdmcp.so.6 (0x00007ff0930e6000)
	libpcre.so.3 => /lib/x86_64-linux-gnu/libpcre.so.3 (0x00007ff09306e000)
	libicudata.so.70 => /lib/x86_64-linux-gnu/libicudata.so.70 (0x00007ff091450000)
	libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007ff091224000)
	libbrotlicommon.so.1 => /lib/x86_64-linux-gnu/libbrotlicommon.so.1 (0x00007ff091201000)
	libbsd.so.0 => /lib/x86_64-linux-gnu/libbsd.so.0 (0x00007ff0911e9000)
	libmd.so.0 => /lib/x86_64-linux-gnu/libmd.so.0 (0x00007ff0911da000)
```

by default, imagick is liked as

```
gcc -shared  .libs/imagick_file.o .libs/imagick_class.o .libs/imagickdraw_class.o .libs/imagickpixel_class.o .libs/imagickpixeliterator_class.o .libs/imagick_helpers.o .libs/imagick.o .libs/imagickkernel_class.o .libs/shim_im6_to_im7.o  -Wl,--rpath -Wl,/usr/local/pkg/gcc-13.2.0/lib/../lib64 -Wl,--rpath -Wl,/usr/local/pkg/gcc-13.2.0/lib/../lib64  /usr/lib/x86_64-linux-gnu/libMagickWand-6.Q16.so /usr/lib/x86_64-linux-gnu/libMagickCore-6.Q16.so  -Wl,-soname -Wl,imagick.so -o .libs/imag
```
however if I remove libgomp from the link statement, everything works,no more SIGSEGV

```
gcc -shared  .libs/imagick_file.o .libs/imagick_class.o .libs/imagickdraw_class.o .libs/imagickpixel_class.o .libs/imagickpixeliterator_class.o .libs/imagick_helpers.o .libs/imagick.o .libs/imagickkernel_class.o .libs/shim_im6_to_im7.o  -Wl,--rpath -Wl,/usr/local/pkg/gcc-13.2.0/lib/../lib64 -Wl,--rpath -Wl,/usr/local/pkg/gcc-13.2.0/lib/../lib64  /usr/lib/x86_64-linux-gnu/libMagickWand-6.Q16.so /usr/lib/x86_64-linux-gnu/libMagickCore-6.Q16.so  -Wl,-soname -Wl,imagick.so -o .libs/imag
cp .libs/imag /usr/local/pkg/php-8.2.10/lib/php/extensions/no-debug-non-zts-20220829/imagick.so
```
Not sure why imagick is linked with gomp. I do not see any openmp code, other than the call to omp_pause_resource_all()

**Comment by bahag-schlachterk** [comment]
I just retried with the ideas by @valerioa, after the latest upgrade to Ubuntu and phpbrew but I found no difference, which makes sense, because I was not able to find a difference in the commands provided. (except the outfile name).

**Comment by valerioa** [comment]
I might have pasted the wrong line. Here are complete directions:

1. Download the imagick extension source code directly, do not use phpbrew.
2. unpack imagick
3. phpize
4. ./configure
5. make
By default, the imagick.so extension will be linked with
```
cc -shared  .libs/imagick_file.o .libs/imagick_class.o \
    .libs/imagickdraw_class.o .libs/imagickpixel_class.o \
    .libs/imagickpixeliterator_class.o .libs/imagick_helpers.o \
    .libs/imagick.o .libs/imagickkernel_class.o .libs/shim_im6_to_im7.o \
    -Wl,--rpath -Wl,/usr/local/pkg/gcc-13.2.0/lib/../lib64 \
    -Wl,--rpath -Wl,/usr/local/pkg/gcc-13.2.0/lib/../lib64 \
    /usr/local/pkg/gcc-13.2.0/lib/../lib64/libgomp.so \
    /usr/lib/x86_64-linux-gnu/libMagickWand-6.Q16.so \
    /usr/lib/x86_64-linux-gnu/libMagickCore-6.Q16.so  \
    -Wl,-soname -Wl,imagick.so -o .libs/imagick.so
```
The statement that links the extension with libgomp is `/usr/local/pkg/gcc-13.2.0/lib/../lib64/libgomp.so` (I use a non standard gcc that I compile myself, your line might be slightly different).

remove that line and re-link with

```
cc -shared  .libs/imagick_file.o .libs/imagick_class.o \
    .libs/imagickdraw_class.o .libs/imagickpixel_class.o \
    .libs/imagickpixeliterator_class.o .libs/imagick_helpers.o \
    .libs/imagick.o .libs/imagickkernel_class.o .libs/shim_im6_to_im7.o \
    -Wl,--rpath -Wl,/usr/local/pkg/gcc-13.2.0/lib/../lib64 \
    -Wl,--rpath -Wl,/usr/local/pkg/gcc-13.2.0/lib/../lib64 \
    /usr/lib/x86_64-linux-gnu/libMagickWand-6.Q16.so \
    /usr/lib/x86_64-linux-gnu/libMagickCore-6.Q16.so  \
    -Wl,-soname -Wl,imagick.so -o .libs/imagick.so
```
Then copy the .libs/imagick.so to the directory where you load your php extensions from. The extension linked without libgomp works great for me.

**Comment by bahag-schlachterk** [comment]
Ok I tried your suggested fix, but the issue seems to be a harder one than I anticipated.

Here is what i tried:
- Downloaded the extension in version 3.7.0
- unpacked
- phpize
- configure
- make
- rerun withtout -lgomp.so
- copy over the created imagick.so file to /.phpbrew/php/php-8.3.0/lib/php/extensions/no-debug-non-zts-20230831

Result remains the same :(
=> segfault after switching to php-8.3.0 => disable imagick and everything works again ...

I am not an expert, but as compiling php in different versions in a docker-container based on alpine works with imagick without any issue, it appears to me that the version of the ImageMagick libs shipped with the most recent version of Ubuntu is somewhat broken.
If I find the time I will may try to setup a vm with a native debian or other distribution to see if the issue can be reproduced there.

**Comment by valerioa** [comment]
Which compiler are you using? 
Can you paste the output of 
gcc --version
?

I suspect changes were made in the latest version of gcc with regard to libgomp and OpenMP. I suspect those changes might impact the imagick plugin

**Comment by valerioa** [comment]
I had a chance to compile php 8.3.0 today. I see the same behavior:
1. download imagick-3.7.0 source
2. unpack
3. cd to imagick-3.7.0
4. phpize
5. ./configure
6. make 
7. make install

the last compilation action is
```
gcc -shared  .libs/imagick_file.o .libs/imagick_class.o .libs/imagickdraw_class.o .libs/imagickpixel_class.o .libs/imagickpixeliterator_class.o .libs/imagick_helpers.o .libs/imagick.o .libs/imagickkernel_class.o .libs/shim_im6_to_im7.o  -Wl,--rpath -Wl,/usr/local/pkg/gcc-13.2.0/lib/../lib64 -Wl,--rpath -Wl,/usr/local/pkg/gcc-13.2.0/lib/../lib64 /usr/local/pkg/gcc-13.2.0/lib/../lib64/libgomp.so /usr/lib/x86_64-linux-gnu/libMagickWand-6.Q16.so /usr/lib/x86_64-linux-gnu/libMagickCore-6.Q16.so  -Wl,-soname -Wl,imagick.so -o .libs/imagick.so
```
With the module linked that way, php will crash. 

remove `/usr/local/pkg/gcc-13.2.0/lib/../lib64/libgomp.so` from the link statement and run

```
gcc -shared  .libs/imagick_file.o .libs/imagick_class.o .libs/imagickdraw_class.o .libs/imagickpixel_class.o .libs/imagickpixeliterator_class.o .libs/imagick_helpers.o .libs/imagick.o .libs/imagickkernel_class.o .libs/shim_im6_to_im7.o  -Wl,--rpath -Wl,/usr/local/pkg/gcc-13.2.0/lib/../lib64 -Wl,--rpath -Wl,/usr/local/pkg/gcc-13.2.0/lib/../lib64  /usr/lib/x86_64-linux-gnu/libMagickWand-6.Q16.so /usr/lib/x86_64-linux-gnu/libMagickCore-6.Q16.so  -Wl,-soname -Wl,imagick.so -o .libs/imagick.so
```
Copy .libs/imagick.so to your extension directory

php works again.

**Comment by oreo639** [comment]
For the backlog, can you install the debug symbols for libgomp?
You can see here for how to do it on Ubuntu:
https://wiki.ubuntu.com/Debug%20Symbol%20Packages

I was able to reproduce the crash on Void, with the cash occurring in `initialize_env()` due to the code assuming `environ` can never be NULL and it would just be reset to a pointer to NULL.

This is the backtrace I got:
```
Program received signal SIGSEGV, Segmentation fault.
0x00007ffff6e1242e in initialize_env () at ./libgomp/env.c:2062
warning: 2062	./libgomp/env.c: No such file or directory
(gdb) bt
#0  0x00007ffff6e1242e in initialize_env () at ./libgomp/env.c:2062
#1  0x00007ffff7fcfffe in call_init (env=0x5555566352b0, argv=0x7fffffffe4c8, 
    argc=1, l=<optimized out>) at ./elf/dl-init.c:70
#2  call_init (l=<optimized out>, argc=1, argv=0x7fffffffe4c8, 
    env=0x5555566352b0) at ./elf/dl-init.c:26
#3  0x00007ffff7fd00e4 in _dl_init (main_map=0x5555566cc1f0, argc=1, 
    argv=0x7fffffffe4c8, env=0x5555566352b0) at ./elf/dl-init.c:117
#4  0x00007ffff7fcc516 in __GI__dl_catch_exception (
    exception=exception@entry=0x0, 
    operate=operate@entry=0x7ffff7fd69c0 <call_dl_init>, 
    args=args@entry=0x7fffffffc9a0) at ./elf/dl-catch.c:211
#5  0x00007ffff7fd695e in dl_open_worker (a=a@entry=0x7fffffffcb40)
    at ./elf/dl-open.c:810
#6  0x00007ffff7fcc489 in __GI__dl_catch_exception (
    exception=exception@entry=0x7fffffffcb20, 
    operate=operate@entry=0x7ffff7fd68d0 <dl_open_worker>, 
    args=args@entry=0x7fffffffcb40) at ./elf/dl-catch.c:237
#7  0x00007ffff7fd6cd8 in _dl_open (
    file=0x7ffff6609000 "/usr/lib/php8.3/modules/imagick.so", 
    mode=<optimized out>, 
    caller_dlopen=0x5555557ac36a <php_load_extension+554>, 
    nsid=<optimized out>, argc=1, argv=0x7fffffffe4c8, env=0x5555566352b0)
    at ./elf/dl-open.c:886
```

Apparently, that was fixed upstream and backported to the gcc-13 branch: https://github.com/gcc-mirror/gcc/commit/c128ad8e830e90a429eaeccc3fb000a73fd6779c

**Comment by Danack** [comment]
@oreo639 Yes, it looks like the cause of this issue is the same as the one fixed in that link, and probably the same cause as fixed in issue https://github.com/Imagick/imagick/issues/624

To be clear, this was an issue in GCC so outside of my control, and not something I could do much about.

If anyone still has an issue when using the latest gcc, please open a new issue.
